Framework/Reflectometry/test/ReflectometryReductionOneAuto3Test.h

```diff
@@ -687,8 +687,39 @@ class ReflectometryReductionOneAuto3Test : public CxxTest::TestSuite {
     TS_ASSERT(AnalysisDataService::Instance().doesExist("IvsQ_binned"));
     TS_ASSERT(AnalysisDataService::Instance().doesExist("IvsQ"));
     TS_ASSERT(AnalysisDataService::Instance().doesExist("IvsLam"));
+  }
 
-    AnalysisDataService::Instance().clear();
+  void test_autodetect_on_inter_instrument_sets_polynomial_correction() {
+    auto inter = loadRun("INTER00013460.nxs");
+    auto const polStringInter =
+        std::string("35.5893,-24.5591,9.20375,-1.89265,0.222291,-0.0148746,0.00052709,-7.66807e-06");
+
+    ReflectometryReductionOneAuto3 alg;
+    alg.initialize();
+    alg.setProperty("InputWorkspace", inter);
+    alg.setProperty("CorrectionAlgorithm", "AutoDetect");
+    alg.execute();
+
+    TS_ASSERT_EQUALS(alg.getPropertyValue("Polynomial"), polStringInter);
+  }
+
+  void test_autodetect_on_surf_instrument_sets_exponential_correction() {
+    // SURF instrument has exponential corrections
+    LoadEmptyInstrument loadInstAlg;
```

Comment:
Thanks for letting me know about this. 
I have added two `Refl_Parameters_[ExponentialCorrection|PolynomialCorrection].xml` files.  
As we discussed on Friday, adding the files was not without consequence, as the instrument cache does not reset and it affects other tests that require the artificial 'REFL' instrument. 
I have implemented the solution we discussed, adding a clear cache method on teardown. But the test would also be fine if we just append the two new test at the end of the test suite file (as test are running in top-down order) to reduce some overhead. 
Let me know what you prefer.

Reply:
Thanks, I guess that makes sense that it would work if we re-ordered the tests. My feeling is that clearing the cache is still a good solution, though, even though it has additional overhead. That way we're not relying on the order of the tests being maintained in future (i.e. it won't matter if someone adds a new test at the end of the file), and it also means that the tests will be robust if they are ever run out of their normal order.